### DISCUSSION TITLE
problems about converting cp2k results to datasets by using cp2kdata

### DISCUSSION THREAD

**Discussion by dlProjectYue** [deepmodeling/dpdata#756] [Q&A] [answered]
When I use cp2kdata to convert the cp2k results, I get the wrong results:
![Weixin Image_20241122174624](https://github.com/user-attachments/assets/159268b7-3f2f-462d-b3eb-0e4bbd388cfc)
I don't know how to fix this. Please help me!

**Comment by robinzyb** [comment]
could you please provide the script?

**Reply by dlProjectYue** [reply]
[md.zip](https://github.com/user-attachments/files/17878423/md.zip)
Hello, this is the script, I hide the cell structure and remain the output information.

**Reply by dlProjectYue** [reply]
I have just tried. If I add       
      ADD_LAST      NUMERIC
      FILENAME      pressure
      LOG_PRINT_KEY TRUE
      in $PRINT$, it shows this wrong information. But I don't konw why? Could you please explain
it?

**Reply by dlProjectYue** [reply]
And another problem is that, when I tried to run 1step MD by excluding above script and use dpdata. It shows wrong information bellow: 
![Weixin Image_20241123144209](https://github.com/user-attachments/assets/ad9b4b27-47a3-4687-adb5-687638e63f5d)
I searched about similar solution which you said that I should delete the energy file. But actually I don't know why this happened. I run 1 step MD, but energy file shows two results. which line in scripts caused this?  Could you please help me?

**Reply by dlProjectYue** [reply]
When I used this md.inp, it still shows AssertionError, assert first_cell.shape==(1,3,3). I don't know why.
[md (2).zip](https://github.com/user-attachments/files/17878709/md.2.zip)

**Reply by dlProjectYue** [reply]
> And another problem is that, when I tried to run 1step MD by excluding above script and use dpdata. It shows wrong information bellow: ![Weixin Image_20241123144209](https://private-user-images.githubusercontent.com/85020793/389175768-ad9b4b27-47a3-4687-adb5-687638e63f5d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzNDg3NTQsIm5iZiI6MTczMjM0ODQ1NCwicGF0aCI6Ii84NTAyMDc5My8zODkxNzU3NjgtYWQ5YjRiMjctNDdhMy00Njg3LWFkYjUtNjg3NjM4ZTYzZjVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIzVDA3NTQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2MzlhNmM4ODI0ZTY2YThiMWNiYjQ2YTNkNDUyZTcxZmE3NzZhNjJlNGJiZjZkNDA3YWFkZTc2NmYwNzYzMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yjJjjem_UCH7ThPzsDhvZVkYdapLKxQqOZEoYgLO320) I searched about similar solution which you said that I should delete the energy file. But actually I don't know why this happened. I run 1 step MD, but energy file shows two results. which line in scripts caused this? Could you please help me?

I know that this is due to the output frequency. If I set printing frequency of different parameters as 1, it didn't show this information.

**Reply by robinzyb** [reply]
> could you please provide the script?

The python code you used to covert cp2k data

**Reply by dlProjectYue** [reply]
I just use codes like "data=dpdata.LabeledSystem('./',cp2k_output_name='output_file.log',fmt='cp2kdata/md')"

**Comment by robinzyb** [comment]
There are some errors on parsing the output. I would like to request you to upload the output in order to diagnise.

**Reply by dlProjectYue** [reply]
[output_file.zip](https://github.com/user-attachments/files/17912465/output_file.zip)
Sorry to bother you. This is my output file.

**Reply by robinzyb** [reply]
I checked the output and found duplicated headers because you submitted multiple times and the outputs went to the same file. I suggest to try
```python
# input your real cell below.
cells = np.arrary([[10.0, 0, 0], [0,10,0], [0,0,10]])
data=dpdata.LabeledSystem('./',cp2k_output_name='output_file.log', cells=cells, fmt='cp2kdata/md')

```

**Reply by dlProjectYue** [reply]
Thanks a lot for you! I have tried your suggestion, but it didn't work. I know the error is caused by duplicated outputs in a file. I tried to delete the output file and run the cp2k again, and the dpdata can read the file successfully. Thanks for your kind help!

**Reply by robinzyb** [reply]
actually you can move the output out of the directory only. That will also work

**Reply by dlProjectYue** [reply]
But if I move out of the output_file.log, what I should fill in 'cp2k_output_name' ?

**Reply by robinzyb** [reply]
it can be None.

**Reply by dlProjectYue** [reply]
Sorry，Like this？ "data=dpdata.LabeledSystem('./',cp2k_output_name='',fmt='cp2kdata/md')". But it shows I cant do like this.

**Reply by robinzyb** [reply]
cp2k_output_name=None

**Reply by dlProjectYue** [reply]
Thanks for your suggestion! I tried to use data=dpdata.LabeledSystem(cp2k_output_name=None) , it shows no error. But data actually has 0 frame.

**Reply by robinzyb** [reply]
supposed to be the following,
```
cells = np.arrary([[10.0, 0, 0], [0,10,0], [0,0,10]])
data=dpdata.LabeledSystem('./',cp2k_output_name=None, cells=cells, fmt='cp2kdata/md')
```
because the cells information is expected to be provided by users instead of `output`
anyhow, glad to see you solve the problem. 

One last thing, once everything's sorted, could you mark one reply as the answer and close this discussion? Thank you and have a good day.

**Reply by dlProjectYue** [reply]
Sorry to bother you again. Even though I know the results, I tried to use your suggestion but it shows the errors like this.
![error](https://github.com/user-attachments/assets/87bfed7b-ebb3-4c76-914f-7c7f491260e5)

**Reply by robinzyb** [reply]
this means the cp2kdata and dpdata are not installed under same environment

**Comment by qizh5** [comment]
Sorry, I met the same question with [dlProjectYue], and I try to delete output.out file .
And the python code is 
cells = np.array([[16.0, 0, 0], [0,16,0], [0,0,16]])
data=dpdata.LabeledSystem('./',cp2k_output_name=None, cells=cells, fmt='cp2kdata/md')
It gave the following error message.
![7f4781075e5245a2324af0c0d41e4f3](https://github.com/user-attachments/assets/d5e79d1d-e59a-42bf-a297-636529a1ff00)
 It seems that an output file must be provided. I don't konw what the problem is. Could you help me?

**Reply by qizh5** [reply]
It's  my output file
[Uploading output.zip…]()

**Reply by robinzyb** [reply, marked as answer]
This is a bug. It is caused by the repetitive information from 
```
 CELL_TOP| Volume [angstrom^3]:                                         4096.000
 CELL_TOP| Vector a [angstrom    16.000     0.000     0.000    |a| =      16.000
 CELL_TOP| Vector b [angstrom     0.000    16.000     0.000    |b| =      16.000
 CELL_TOP| Vector c [angstrom     0.000     0.000    16.000    |c| =      16.000
 CELL_TOP| Angle (b,c), alpha [degree]:                                   90.000
 CELL_TOP| Angle (a,c), beta  [degree]:                                   90.000
 CELL_TOP| Angle (a,b), gamma [degree]:                                   90.000
 CELL_TOP| Numerically orthorhombic:                                         YES
 
 GENERATE|  Preliminary Number of Bonds generated:                             0
 GENERATE|  Achieved consistency in connectivity generation.

 CELL| Volume [angstrom^3]:                                             4096.000
 CELL| Vector a [angstrom]:      16.000     0.000     0.000    |a| =      16.000
 CELL| Vector b [angstrom]:       0.000    16.000     0.000    |b| =      16.000
 CELL| Vector c [angstrom]:       0.000     0.000    16.000    |c| =      16.000
 CELL| Angle (b,c), alpha [degree]:                                       90.000
 CELL| Angle (a,c), beta  [degree]:                                       90.000
 CELL| Angle (a,b), gamma [degree]:                                       90.000
 CELL| Numerically orthorhombic:                                             YES

 CELL_REF| Volume [angstrom^3]:                                         4096.000
 CELL_REF| Vector a [angstrom    16.000     0.000     0.000    |a| =      16.000
 CELL_REF| Vector b [angstrom     0.000    16.000     0.000    |b| =      16.000
 CELL_REF| Vector c [angstrom     0.000     0.000    16.000    |c| =      16.000
 CELL_REF| Angle (b,c), alpha [degree]:                                   90.000
 CELL_REF| Angle (a,c), beta  [degree]:                                   90.000
 CELL_REF| Angle (a,b), gamma [degree]:                                   90.000
 CELL_REF| Numerically orthorhombic:                                         YES
```
in `output`.

The quick way to resolve the problem is deleting extra CELL infomation in the output and keeping only one cell infomation. You could try this first.
I will fix the bug later on.

**Reply by qizh5** [reply]
> Excuse me, may I ask for detailed information on which part to delete, cell_top,cell,or cell_ref? It should be okay,if i only keep the following part? CELL_TOP| Volume [angstrom^3]: 4096.000 CELL_TOP| Vector a [angstrom 16.000 0.000 0.000 |a| = 16.000 CELL_TOP| Vector b [angstrom 0.000 16.000 0.000 |b| = 16.000 CELL_TOP| Vector c [angstrom 0.000 0.000 16.000 |c| = 16.000 CELL_TOP| Angle (b,c), alpha [degree]: 90.000 CELL_TOP| Angle (a,c), beta [degree]: 90.000 CELL_TOP| Angle (a,b), gamma [degree]: 90.000 CELL_TOP| Numerically orthorhombic: YES
> 
> GENERATE| Preliminary Number of Bonds generated: 0 GENERATE| Achieved consistency in connectivity generation.

Excuse me, may I ask for detailed information on which part to delete, cell_top,cell,or cell_ref?
It should be okay,if i only keep the following part?
CELL_TOP| Volume [angstrom^3]: 4096.000
CELL_TOP| Vector a [angstrom 16.000 0.000 0.000 |a| = 16.000
CELL_TOP| Vector b [angstrom 0.000 16.000 0.000 |b| = 16.000
CELL_TOP| Vector c [angstrom 0.000 0.000 16.000 |c| = 16.000
CELL_TOP| Angle (b,c), alpha [degree]: 90.000
CELL_TOP| Angle (a,c), beta [degree]: 90.000
CELL_TOP| Angle (a,b), gamma [degree]: 90.000
CELL_TOP| Numerically orthorhombic: YES

GENERATE| Preliminary Number of Bonds generated: 0
GENERATE| Achieved consistency in connectivity generation.

**Reply by robinzyb** [reply]
You will find the part I quote is repetitive in the output. Just keep one of them.

**Reply by qizh5** [reply]
> You will find the part I quote is repetitive in the output. Just keep one of them.

Got it!!!!Thanks for your teaching ,now it works well!!!!!

**Reply by robinzyb** [reply]
glad you have solve it!
